src/components/MapVizzard/MapOptions/OptionsMask.jsx

```diff
@@ -0,0 +1,268 @@
+import OLVectorLayer from 'ol/layer/Vector';
+import Slider from '@mui/material/Slider';
+import { createTheme } from '@mui/material/styles';
+import grey from '@mui/material/colors/grey';
+import { Draw, Modify, Snap } from 'ol/interaction';
+import MultiPoint from 'ol/geom/MultiPoint';
+import WKT from 'ol/format/WKT';
+import Feature from 'ol/Feature';
+import { Vector as VectorSource } from 'ol/source';
+import {
+  Style, Fill, Stroke, Circle,
+} from 'ol/style';
+import { transform } from 'ol/proj';
+import TextField from '@mui/material/TextField';
+import {
+  FormControl,
+} from '@mui/material';
+import styles from './MapOptions.module.css';
+import mask from '../assets/mask.svg';
+
+function OptionsMask({
+  layer, activeLayer, updateLayer, map,
+}) {
+  const theme = createTheme({
+    palette: {
+      primary: grey,
+    },
+  });
+  const layerClone = { ...layer };
```

Comment:
Instead of cloning the element at the top. It's better to only clone before mutation.

Ideally we should not share reference of the object that we will mutate.